receiver/azureblobrehydrationreceiver/receiver.go

```diff
@@ -186,22 +190,31 @@ func (r *rehydrationReceiver) scrape() {
 	checkpoint := r.loadCheckpoint(r.ctx)
 
 	// Call once before the loop to ensure we do a collection before the first ticker
-	r.rehydrateBlobs(checkpoint, marker)
+	numBlobsRehydrated := r.rehydrateBlobs(checkpoint, marker)
+	emptyBlobCounter := checkBlobCount(numBlobsRehydrated, 0)
 
 	for {
 		select {
 		case <-r.ctx.Done():
 			return
 		case <-ticker.C:
-			r.rehydrateBlobs(checkpoint, marker)
+			// Polling for blobs has egress charges so we want to stop polling
+			// after we stop finding blobs.
+			if emptyBlobCounter == emptyPollLimit {
+				continue
+			}
```

Comment:
Does it make sense in this case to just return and break out of this loop and kill this goroutine?

Reply:
how does the collector handle shutting down individual components? that seems like it might cause problems.

Reply:
I thought about this and was unsure. I think it's probably ok just to break out. The scrape function is private anyways.

Reply:
The collector calls the Shutdown function to shut down the component. It shouldn't cause any problems to exit this goroutine, it'll just free up the resources used here (goroutine and ticker).